gips/data/landsat/landsat.py

```diff
@@ -682,11 +682,7 @@ def process(self, products=None, overwrite=False, **kwargs):
 
         start = datetime.now()
 
-        assets = set() # assets needed for this process() run
-        for key, val in products.requested.items():
-            assets.update(self._products[val[0]]['assets'])
```

Comment:
Now that I see this change again, I'm questioning my thought that we could assume asset presence/abscence has already been handled.  I'm thinking backing this part out, and using the slightly better name of `requested_assets` instead of `assets` might be needed.  let's discuss on Monday morning.

Reply:
So I guess `self.assets` is populated in `Data.__init__` by `[self.add_asset(a) for a in self.Asset.discover(tile, date)]`.  But that could find no assets in it if none are discovered, I think.

Reply:
I could change the check o nline 680 to `if 0 in (len(products), len(self.assets)): return`; what do you think?

Reply:
That plus high-verbosity logging of the absence of an asset to process (maybe verbosity 5?)